File: insertion/cpp/insertionNums.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <time.h>

int main(int argc, char *argv[2])
{
    std::ifstream input("numeros 1.txt");
    std::ofstream output("./insCppNums.txt", std::ios_base::app);
    std::vector<int> vetor;
    int aux;
    unsigned long comp = 0, swap = 0;
    clock_t inicio, fim;

    for (int i = 0; i < std::atoi(argv[1]); i++)
    {
        input >> aux;
        vetor.push_back(aux);
    }

    inicio = clock();

    for (int i = 0; i < std::size(vetor)-1; i++)
    {
        for (int j = i + 1; j > 0; j--)
        {   
            comp++;
            if (vetor[j] < vetor[j - 1])
            {
                aux = vetor[j];
                vetor[j] = vetor[j - 1];
                vetor[j - 1] = aux;
                swap++;
            }
            else
                break;
        }
    }

    fim = clock();
    output << argv[1] << "," << (float)(((fim - inicio) + 0.0) / CLOCKS_PER_SEC) << "," << comp << "," << swap << "\n";

    
    return 0;
}

File: insertion/golang/insertionNums.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

func main() {

	input, err := os.Open("numeros 1.txt")
	if err != nil {
		panic(err)
	}
	defer input.Close()

	output, err := os.OpenFile("./insGoNums.txt", os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
	if err != nil {
		panic(err)
	}
	defer input.Close()
	
	vetor := make([]int, 0)

	var comp, swap int

	num, err := strconv.Atoi(os.Args[1])

	for i := 0; i < num; i++ {
		var aux int
		fmt.Fscan(input, &aux)
		vetor = append(vetor, aux)
	}

	inicio := time.Now()
	
	for i := 0; i < len(vetor)-1; i++ {
		for j := i + 1; j > 0; j-- {
			comp++
			if vetor[j] < vetor[j-1] {
				var aux1 int
				aux1 = vetor[j]
				vetor[j] = vetor[j-1]
				vetor[j-1] = aux1
				swap++
			} else {
				break
			}
		}
	}

	fim := time.Now()

	fmt.Fprintf(output, "%s,%v,%d,%d\n", os.Args[1], float64(fim.Sub(inicio).Seconds()), comp, swap)
}
